lib/Site.js

```diff
@@ -185,6 +189,34 @@ Site.prototype.collectBaseUrl = function () {
   return Promise.resolve();
 };
 
+Site.prototype.collectbaseUrlToDefinedVariablesMap = function () {
+  this.baseUrlToDefinedVariablesMap = {};
+
+  Object.keys(this.baseUrlMap).forEach((base) => {
+    const userDefinedVariables = {};
+    const definedVariablePath = path.resolve(base, USER_VARIABLES_PATH);
+    let content;
+    try {
+      content = fs.readFileSync(definedVariablePath, 'utf8');
+    } catch (e) {
+      content = '';
+      logger.info(e.message);
+    }
+    const $ = cheerio.load(content);
+    $.root().children().each(function () {
+      const id = $(this).attr('id');
+      const html = $(this).html();
+      userDefinedVariables[id] = html;
+    });
+
+    // Currently baseUrl has not been calculated.
+    // This is to prevent the first nunjuck call from converting {{baseUrl}} to nothing
```

Comment:
- ~`Currently`~`At this point,`
- "to ~`nothing`~`an empty string`" feels less spooky.